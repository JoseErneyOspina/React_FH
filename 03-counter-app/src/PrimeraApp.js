import React from 'react';

// FC -> Functional Components
const PrimeraApp = ( props ) => {

    const saludo = 'Hola Mundo const';
    // const saludo = 123;
    // const saludo = 123.45;
    // const saludo = true;
    // const saludo = [1, 2, 3, 4];
    // const saludo = {
    //     nombre: 'Fernando',
    //     edad: 34
    // }

    return (
        <>
            <h1> { saludo } </h1>
            {/* <pre> { JSON.stringify( saludo, null, 3 ) } </pre>  */}
            <p>Mi primera aplicaci√≥n</p>
        </>
    );
};

export default PrimeraApp;