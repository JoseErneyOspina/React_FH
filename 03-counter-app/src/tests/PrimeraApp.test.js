import React from 'react';
// import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import PrimeraApp from '../PrimeraApp';


describe('Testing on <PrimeraApp />', () => {

    // test('should show the message "Soy, Goku"', () => {
    //     const saludo = 'Hola, soy Goku';
    //     const { getByText } = render( <PrimeraApp saludo = { saludo } /> )
    //     expect( getByText( saludo ) ).toBeInTheDocument();
    // });

    test('should show <PrimeraApp /> correctamente', () => {

        const saludo = 'Hola, soy Goku';
        const wrapper = shallow( <PrimeraApp saludo = { saludo } />);

        expect( wrapper ).toMatchSnapshot();
    });

    test('should return the subtitle send by props', () => {

        const saludo = 'Hola, soy Goku';
        const subtitulo = 'Soy un subtitulo'
        const wrapper = shallow(
            <PrimeraApp
                saludo = { saludo }
                subtitulo = { subtitulo }
            />
        );

        const textoParrafo = wrapper.find('p').text();
        expect(textoParrafo).toBe( subtitulo );
    });
});