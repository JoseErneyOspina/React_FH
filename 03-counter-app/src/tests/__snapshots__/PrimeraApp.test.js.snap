// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on <PrimeraApp /> should show <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
     
    Hola, soy Goku
     
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
